sdk/python/kfp/components/_naming.py

```diff
@@ -72,3 +75,27 @@ def _make_name_unique_by_adding_index(name:str, collection, delimiter:str):
             if unique_name not in collection:
                 break
     return unique_name
+
+
+def _convert_name_and_make_it_unique_by_adding_number(name: str, used_converted_names, conversion_func: Callable[[str], str]):
+    converted_name = conversion_func(name)
+    if converted_name in used_converted_names:
+        for i in range(2, sys.maxsize ** 10): #Starting indices from 2: "Something", "Something_2", ...
+            converted_name = conversion_func(name + ' ' + str(i))
+            if converted_name not in used_converted_names:
+                break
+    return converted_name
+
+
+def generate_unique_name_conversion_table(names: Sequence[str], conversion_func: Callable[[str], str]) -> Mapping[str, str]:
```

Comment:
Do you have test cover this method?

Reply:
As this is private API, it does not have direct test cases, but it's covered by the following test:
https://github.com/kubeflow/pipelines/blob/f5bdf2474e1529d2093f8efbb962c8c071718792/sdk/python/tests/components/test_components.py#L156

Reply:
Great, thanks.